src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/CpsPackageReferenceProject.cs

```diff
@@ -172,21 +177,49 @@ private static PackageReference ToPackageReference(LibraryDependency library, Nu
             return new PackageReference(identity, targetFramework);
         }
 
-        public override async Task<Boolean> InstallPackageAsync(PackageIdentity packageIdentity, DownloadResourceResult downloadResourceResult, INuGetProjectContext nuGetProjectContext, CancellationToken token)
+        public override async Task<Boolean> InstallPackageAsync(PackageIdentity packageIdentity,
+            INuGetProjectContext nuGetProjectContext,
+            IEnumerable<NuGetFramework> successfulFrameworks,
+            IEnumerable<NuGetFramework> unsuccessfulFrameworks,
+            CancellationToken token)
         {
-
             nuGetProjectContext.Log(MessageLevel.Info, Strings.InstallingPackage, packageIdentity);
 
-            var configuredProject = await _unconfiguredProject.GetSuggestedConfiguredProjectAsync();
-            var result = await
-                configuredProject?.Services.PackageReferences.AddAsync
-                (packageIdentity.Id, packageIdentity.Version.ToNormalizedString());
-            
-            // This is the update operation
-            if (result != null && !result.Added)
+            if (successfulFrameworks.Any() && unsuccessfulFrameworks.Any())
+            {
+                var conditionalService = _unconfiguredProject
+                    .Services
+                    .ExportProvider
+                    .GetExportedValue<IConditionalPackageReferencesService>();
+
+                if (conditionalService == null)
+                {
+                    return true;
+                }
+
+                foreach (var framework in successfulFrameworks)
+                {
+                    await conditionalService.AddAsync(
+                        packageIdentity.Id,
+                        packageIdentity.Version.ToNormalizedString(),
+                        TargetFrameworkCondition,
+                        framework.GetShortFolderName());
```

Comment:
Ah, good point. I'm not sure how to get the original string from CPS yet. I'll look into it.